host/xtest/regression_4100.c

```diff
@@ -0,0 +1,2536 @@
+		  "-1FFFFFFFFFFFFFFFE" },
+		/* single word, no carry */
+		{ "FFFF", "-FF", "FF00" },
+		/* single word, with carry */
+		{ "F00000000", "-00000FFFF", "EFFFF0001" },
+		/* multi words with carry */
+		{ "FFFFFFFF00000000", "-FFFFFFFF", "FFFFFFFE00000001" },
+		{ "10000000FFFFFFFF00000000", "-FFFFFFFF",
+		  "10000000FFFFFFFE00000001" },
+		{ "FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF",
+		  "-FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFE", "1" },
+		{ "FF", "-FFFF", "-FF00" },
+		{ "FFFFFFFF", "-1FFFFFFFF", "-100000000" },
+		/* single word, no carry */
+		{ "-FFFF", "FF", "-FF00" },
+		/* single word, with carry */
+		{ "-F00000000", "00000FFFF", "-EFFFF0001" },
+		/* multi words with carry */
+		{ "-FFFFFFFF00000000", "FFFFFFFF", "-FFFFFFFE00000001" },
+		{ "-10000000FFFFFFFF00000000", "FFFFFFFF",
+		  "-10000000FFFFFFFE00000001" },
+		{ "-FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF",
+		  "FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFE", "-1" },
+		{ "-FF", "FFFF", "FF00" },
+		{ "-FFFFFFFF", "1FFFFFFFF", "100000000" },
+	};
+
+	if (!ADBG_EXPECT_TEEC_SUCCESS(c,
+		xtest_teec_open_session(&session, &crypt_user_ta_uuid, NULL,
+					&ret_orig)))
+		return;
+
+	for (n = 0; n < ARRAY_SIZE(data); n++) {
+		if (!ADBG_EXPECT_TRUE(c, do_addsub(c, &session, data[n].s,
+				data[n].t, data[n].r))) {
+			Do_ADBG_Log("n %zu", n);
+			break;
+		}
+	}
+
+
+	TEEC_CloseSession(&session);
+}
+ADBG_CASE_DEFINE(regression, 4105, test_4105,
+		"Test TEE Internal API Arithmetical API - Add, Sub");
+
+static void test_4106(ADBG_Case_t *c)
+{
+	TEEC_Session session = { 0 };
+	uint32_t ret_orig;
+	size_t n;
+	uint32_t h1 = TA_CRYPT_ARITH_INVALID_HANDLE;
+	uint32_t h2 = TA_CRYPT_ARITH_INVALID_HANDLE;
+	static const struct {
+		const char *val;
+		const char *neg_val;
+	} data[] = {
+		{ "0", "0" },
+		{ "1", "-1" },
+		{ "123", "-123" },
+		{ "123456789123456789", "-123456789123456789" },
+	};
+
+	if (!ADBG_EXPECT_TEEC_SUCCESS(c,
+		xtest_teec_open_session(&session, &crypt_user_ta_uuid, NULL,
+					&ret_orig)))
+		return;
+
+	if (!ADBG_EXPECT_TEEC_SUCCESS(c, cmd_new_var(c, &session, 1024, &h1)))
+		goto out;
+	if (!ADBG_EXPECT_TEEC_SUCCESS(c, cmd_new_var(c, &session, 1024, &h2)))
+		goto out;
+
+	for (n = 0; n < ARRAY_SIZE(data); n++) {
+		if (!ADBG_EXPECT_TEEC_SUCCESS(c, convert_from_string(c,
+				&session, data[n].val, h1)))
+			goto out;
+
+		if (!ADBG_EXPECT_TEEC_SUCCESS(c, convert_from_string(c,
+				&session, data[n].neg_val, h2)))
+			goto out;
+
+		if (n && !ADBG_EXPECT_TEEC_SUCCESS(c, compare_handle(c,
+				&session, h1, h2, 1)))
+			goto out;
+
+		if (!ADBG_EXPECT_TEEC_SUCCESS(c, cmd_neg(c, &session, h1, h1)))
+			goto out;
+		if (!ADBG_EXPECT_TEEC_SUCCESS(c, compare_handle(c, &session,
+				h1, h2, 0)))
+			goto out;
+	}
+out:
+	TEEC_CloseSession(&session);
+}
+ADBG_CASE_DEFINE(regression, 4106, test_4106,
+		"Test TEE Internal API Arithmetical API - Neg");
+
+static bool do_mul(ADBG_Case_t *c, TEEC_Session *s, const char *str_s,
+		      const char *str_t, const char *str_r)
```

Comment:
few minor indentation issues: some manual space indent that do not match.
Check also definition header for `test_4110_mod()` and `test_4110_fmm()`.